src/library_webgpu.js

```diff
@@ -1180,9 +1188,8 @@ var LibraryWebGPU = {
     abort('TODO: wgpuDeviceCreateComputePipelineAsync unimplemented');
   },
 
-  wgpuDeviceCreateRenderPipeline: (deviceId, descriptor) => {
+  $generateRenderPipelineDesc: (descriptor) => {
```

Comment:
@sbc100 is this the right way to have a "private" helper function?

Reply:
The dollar-sign prefix is for a symbols that are not accessible to native code.   Helpers, and/or funtions that take non-wasm types should all be marked like this.

If you really want to make something internal you can also tag it with `__internal: true`, which means user-supplied libraries JS files also can't link against it.

Reply:
Great, thanks! We don't really `__internal` but it wouldn't hurt, and it would be a little more self-documenting, so I'd say let's add it. I'm assuming that's
`$generateRenderPipelineDesc__internal: true,`

Reply:
Yup

Reply:
Done in 84dae207